pkg/config/trustengine.go

```diff
@@ -1,49 +1,51 @@
 package config
 
 import (
-	"fmt"
+	"errors"
 	"os"
 
 	piperhttp "github.com/SAP/jenkins-library/pkg/http"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/trustengine"
 )
 
-func ResolveAllTrustEngineReferences(config *StepConfig, params []StepParameters, trustEngineConfiguration trustengine.Configuration) {
+// ResolveAllTrustEngineReferences retrieves all the step's secrets from the trust engine
+func ResolveAllTrustEngineReferences(config *StepConfig, params []StepParameters, trustEngineConfiguration trustengine.Configuration, client *piperhttp.Client) {
 	for _, param := range params {
 		if ref := param.GetReference("trustengineSecret"); ref != nil {
 			if config.Config[param.Name] == "" { // what if Jenkins set the secret?
-				resolveTrustEngineReference(ref, config, &piperhttp.Client{}, param, trustEngineConfiguration)
+				log.Entry().Infof("Getting %s from Trust Engine", ref.Name)
+				token, err := trustengine.GetToken(ref.Name, client, trustEngineConfiguration)
+				if err != nil {
+					log.Entry().Info(" failed")
+					log.Entry().WithError(err).Warnf("Couldn't get %s secret from Trust Engine", param.Name)
+					continue
+				}
+				log.RegisterSecret(token)
+				config.Config[param.Name] = token
+				log.Entry().Info(" succeeded")
 			}
 		}
 	}
 }
 
-// resolveTrustEngineReference retrieves a secret from Vault trust engine
-func resolveTrustEngineReference(ref *ResourceReference, config *StepConfig, client *piperhttp.Client, param StepParameters, trustEngineConfiguration trustengine.Configuration) {
-	token, err := trustengine.GetToken(ref.Name, client, trustEngineConfiguration)
-	if err != nil {
-		log.Entry().Infof(fmt.Sprintf("couldn't get secret from trust engine: %s", err))
-		return
-	}
-	log.RegisterSecret(token)
-	config.Config[param.Name] = token
-	log.Entry().Infof("retrieving %s token from trust engine succeeded", ref.Name)
-}
-
 // SetTrustEngineConfiguration sets the server URL and token
-func (c *Config) SetTrustEngineConfiguration(hookConfig map[string]interface{}) {
+func (c *Config) SetTrustEngineConfiguration(hookConfig map[string]interface{}) error {
```

Comment:
Most of the new function are fine with being limited to the package and not being exposed.

Reply:
I've now made some functions private.